Lib/_pyrepl/unix_console.py

```diff
@@ -566,11 +574,9 @@ def clear(self):
 
     @property
     def input_hook(self):
-        try:
-            import posix
-        except ImportError:
-            return None
-        if posix._is_inputhook_installed():
+        # avoid inline imports here so the repl doesn't get flooded with import
+        # logging from -Ximporttime=2
+        if posix is not None and posix._is_inputhook_installed():
             return posix._inputhook
```

Comment:
perhaps use `try`/`except NameError` here, instead of the `None` checks?

Reply:
to me that feels like too broad of an `except`, for example it would suppress a desired exception in the event that the `posix` API changes.

Reply:
The `is not None` check is shorter than a try:except: block. Let's leave as is.